storage/src/key_value/mod.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{any::Any, borrow::Cow};
+
+use tokio::sync::{mpsc, oneshot};
+
+use crate::{Digest, SerialBlock, SerialRecord};
+use anyhow::*;
+
+mod storage;
+pub use storage::*;
+
+mod column;
+pub use column::*;
+
+mod agent;
+use agent::Agent;
+
+pub type Value<'a> = Cow<'a, [u8]>;
+
+pub trait KeyValueStorage {
+    fn get<'a>(&'a mut self, column: KeyValueColumn, key: &[u8]) -> Result<Option<Value<'a>>>;
+
+    fn exists<'a>(&'a mut self, column: KeyValueColumn, key: &[u8]) -> Result<bool>;
+
+    fn get_all_keys<'a>(&'a mut self, column: KeyValueColumn) -> Result<Vec<Value<'a>>>;
+
+    fn get_all<'a>(&'a mut self, column: KeyValueColumn) -> Result<Vec<(Value<'a>, Value<'a>)>>;
```

Comment:
It's still a single column, so maybe `get_whole_column` or `get_column_all`?

Reply:
sounds reasonable